fuse_constraints/src/marginalize_variables.cpp

```diff
@@ -363,6 +363,11 @@ LinearTerm linearize(
 
 LinearTerm marginalizeNext(const std::vector<LinearTerm>& linear_terms)
 {
+  if (linear_terms.empty())
```

Comment:
This is the main change / patch from this PR. The rest are minor changes, provided in two other separate commits.